################################################################################
SHORT DESCRIPTION: 
################################################################################
Avoid panic in cpufreq_gov_stop

################################################################################
LONG DESCRIPTION: 
################################################################################
The problem is that cpufreq_gov_stop is called without having a start first so
this patch will check if the givernor has been enabled before disabling it.

################################################################################
CHANGELOG 
################################################################################

################################################################################
REMOVAL 
################################################################################

################################################################################
UPSTREAM PLAN
################################################################################
Maybe, not sure. We have no repro steps and w/o a repro no one is going to
attempt to upstream it.

################################################################################
INTERNAL DEPENDENCIES 
################################################################################

################################################################################
PATCHES 
################################################################################
Index: xen-4.6.5/xen/drivers/cpufreq/cpufreq_ondemand.c
===================================================================
--- xen-4.6.5.orig/xen/drivers/cpufreq/cpufreq_ondemand.c
+++ xen-4.6.5/xen/drivers/cpufreq/cpufreq_ondemand.c
@@ -273,8 +273,11 @@ int cpufreq_governor_dbs(struct cpufreq_
         break;
 
     case CPUFREQ_GOV_STOP:
-        dbs_timer_exit(this_dbs_info);
-        dbs_enable--;
+        if (this_dbs_info->enable)
+        {
+            dbs_timer_exit(this_dbs_info);
+            dbs_enable--;
+        }
 
         break;
 
